src/core/transaction.c

```diff
@@ -406,6 +424,12 @@ static int transaction_verify_order_one(Transaction *tr, Job *j, Job *from, unsi
                                                     *unit_id, *job_type),
                                    "%s", strna(unit_ids));
 
+                if (dot && IN_SET(log_get_target(), LOG_TARGET_AUTO, LOG_TARGET_JOURNAL, LOG_TARGET_JOURNAL_OR_KMSG)) {
+                        log_struct(LOG_WARNING,
+                                   LOG_UNIT_MESSAGE(j->unit, "Ordering cycle generated, see TRANSACTION_CYCLE= for details. Note that each arc represents Before= dependency."),
+                                   "TRANSACTION_CYCLE=%s", dot);
```

Comment:
Do you have an example of generated output? I'd think that usually the cycle is a simple loop, so the existing output that just lists the units involved is enough?

Reply:
Here is example of super simple cycle, created because of following units,

```
# a.service
[Unit]
Before=b.service
Wants=b.service c.service

[Service]
ExecStart=/bin/true

[Install]
WantedBy=multi-user.target

# b.service
[Unit]
Before=c.service

[Service]
ExecStart=/bin/true

[Install]
WantedBy=multi-user.target

# c.service
[Unit]
Before=a.service

[Service]
ExecStart=/bin/true

[Install]
WantedBy=multi-user.target
```

![cycle](https://github.com/user-attachments/assets/de9d52df-9649-41ea-90c7-65095a59168c)